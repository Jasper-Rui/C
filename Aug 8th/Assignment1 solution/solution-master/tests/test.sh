#!/bin/sh

echo ======================================================
echo ====================== TEST $1 ========================
echo ======================================================
if timeout 10 ./miner/miner < tests/test.$1.in > tests/test.$1.out; then 
  if diff tests/test.$1.out tests/test.$1.expected > /dev/null; then
    echo PASSED
  else
    echo FAILED
    echo ======
    echo Expected Output:
    echo ++++++++++++++++
    cat tests/test.$1.expected
    echo =====================
    echo Actual Output:
    echo ++++++++++++++++
    cat tests/test.$1.out
    exit 1
  fi
elif [ $? == 124 ]; then
  echo TIMEOUT
  exit 1
else 
  echo Abnormal program termination: the program crashed
  echo Exit code $?
  exit 1
fi
