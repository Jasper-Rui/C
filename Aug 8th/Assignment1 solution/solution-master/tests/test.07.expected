Adding transaction: 11235 Alice_Q_Public Bob_X_Private 150000 20
Adding transaction: 12358 Carol_Y_Incredible Dave_Z_Particle 350000 40
Block mined: 1 0 0x00000000 2
11235 Alice_Q_Public Bob_X_Private 150000 20
12358 Carol_Y_Incredible Dave_Z_Particle 350000 40
0x00000055 0x00cdc610
